### PR TITLE
UF-7195: Added resource for letter statistics by department

### PR DESCRIPTION
**Pull request by MickeWallner**

### CODE REVIEW

**Review by cheezi0747 — DISMISSED** [review]

File: src/main/java/se/sundsvall/messaging/api/StatusAndHistoryResource.java [hunk 1]
```diff
@@ -322,14 +321,51 @@ ResponseEntity<Statistics> getStats(
...
+                description = "Internal Server Error",
+                content = @Content(schema = @Schema(implementation = Problem.class))
+            )
+        }
+    )
+    @GetMapping(
+        value = {"/statistics/departments","/statistics/departments/{department}"},
+        produces = { APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE }
+    )
+    ResponseEntity<DepartmentStatistics> getDepartmentStats(
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/messaging/api/StatusAndHistoryResource.java:344]: Is it correct to use @ArraySchema? The return type is not a list. Should maybe just set "useReturnTypeSchema = true" instead?

**Review by cheezi0747 — APPROVED** [review]

File: src/test/java/se/sundsvall/messaging/api/StatusAndHistoryResourceTests.java [hunk 2]
```diff
@@ -188,4 +189,24 @@ void test_toHistoryResponse(final MessageType messageType) {
         assertThat(result.status()).isEqualTo(SENT);
         assertThat(result.timestamp()).isNotNull();
     }
+
+    @Test
```
Comment (josc04) [inline src/test/java/se/sundsvall/messaging/api/StatusAndHistoryResourceTests.java:193]: Add tests to also verify that sending null values works as intended?

File: src/main/java/se/sundsvall/messaging/api/StatusAndHistoryResource.java [hunk 3]
```diff
@@ -322,14 +321,51 @@ ResponseEntity<Statistics> getStats(
...
```
Comment (josc04) [inline src/main/java/se/sundsvall/messaging/api/StatusAndHistoryResource.java:344]: This will allow empty string as value for department, from and to, which might not be desirable? Maybe add validation that checks that value is null or not empty (not sure if such annotation exists or if it needs to be created, but maybe regex-annotation can be used if it considers null as valid).
